testing/dart/observatory/tracing_test.dart

```diff
@@ -65,6 +65,6 @@ void main() {
       }
     }
     expect(saveLayerRecordCount, 3);
-    expect(saveLayerCount, 3);
+    expect(saveLayerCount, 6);
```

Comment:
Why did this change from 3 to 6? That seems odd, like we are getting double-accounting here.

Reply:
No, actually, it is because the leaf_nodes_builder is not set so DisplayListLayer::Paint will replay its DL into the flattened DL rather than inserting it by reference. Then later the flattened DL will replay the saveLayer calls a second time and they will be double-accounted.